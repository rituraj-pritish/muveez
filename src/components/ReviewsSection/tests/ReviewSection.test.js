import userEvent from '@testing-library/user-event'
import React from 'react'
import { render, screen } from 'tests/utils'
import ReviewSection from '../index'

describe('Reviews section', () => {
	test('Render empty message if no data', () => {
		const { container } = render(<ReviewSection data={[]} />)
		expect(screen.getByText('Reviews')).toBeInTheDocument()
		expect(screen.getByText('No reviews available')).toBeInTheDocument()

		expect(container).toMatchSnapshot()
	})

	test('Does not Render view all link', () => {
		const { container } = render(<ReviewSection data={{}, {}} />)
		expect(screen.queryByText('View all')).not.toBeInTheDocument()

		expect(container).toMatchSnapshot()
	})

	test('Render view all link if totals reviews greater than 3', () => {
		const { container, history } = render(
			<ReviewSection
				data={[{}, {}, {}, {}]}
				viewAllLink='/link'
			/>
		)
		expect(screen.getByText('View all')).toBeInTheDocument()

		jest.spyOn(history, 'push')
		userEvent.click(screen.getByText('View all'))
		expect(history.push).toHaveBeenCalledWith('/link')

		expect(container).toMatchSnapshot()
	})
})
