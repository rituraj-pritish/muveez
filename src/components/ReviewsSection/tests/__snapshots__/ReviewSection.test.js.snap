// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews section Does not Render view all link 1`] = `
<div>
  <div
    class="css-ahmskn-SkeletonTheme"
  >
    <div>
      <div
        class="sc-dkPtyc ggyVGj"
      >
        <div
          class="sc-hKwCoD ecqkwW"
          color="textPrimary"
          font-size="4"
        >
          Reviews
        </div>
      </div>
      <div
        class="sc-dkPtyc dxBLBq"
        height="50"
      >
        No reviews available
      </div>
    </div>
  </div>
</div>
`;

exports[`Reviews section Render empty message if no data 1`] = `
<div>
  <div
    class="css-ahmskn-SkeletonTheme"
  >
    <div>
      <div
        class="sc-dkPtyc ggyVGj"
      >
        <div
          class="sc-hKwCoD ecqkwW"
          color="textPrimary"
          font-size="4"
        >
          Reviews
        </div>
      </div>
      <div
        class="sc-dkPtyc dxBLBq"
        height="50"
      >
        No reviews available
      </div>
    </div>
  </div>
</div>
`;

exports[`Reviews section Render view all link if totals reviews greater than 3 1`] = `
<div>
  <div
    class="css-ahmskn-SkeletonTheme"
  >
    <div>
      <div
        class="sc-dkPtyc ggyVGj"
      >
        <div
          class="sc-hKwCoD ecqkwW"
          color="textPrimary"
          font-size="4"
        >
          Reviews
        </div>
        <div
          class="sc-hKwCoD bVSAYo"
          color="secondary"
          font-size="1"
        >
          <a
            class="sc-eCImvq iuTZXO"
            href="/link"
          >
            View all
          </a>
        </div>
      </div>
      <div
        class="sc-dPiKHq ktXSgo"
      >
        <div
          class="sc-dkPtyc gzMWFX"
        >
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF dkWHGm css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 40px; height: 40px; border-radius: 50%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF evKoAY css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 150px; height: 20px;"
            >
              ‌
            </span>
          </span>
        </div>
        <span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="sc-dPiKHq ktXSgo"
      >
        <div
          class="sc-dkPtyc gzMWFX"
        >
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF dkWHGm css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 40px; height: 40px; border-radius: 50%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF evKoAY css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 150px; height: 20px;"
            >
              ‌
            </span>
          </span>
        </div>
        <span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="sc-dPiKHq ktXSgo"
      >
        <div
          class="sc-dkPtyc gzMWFX"
        >
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF dkWHGm css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 40px; height: 40px; border-radius: 50%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton sc-jRQAMF evKoAY css-1eg0sfo-skeletonStyles-Skeleton"
              style="width: 150px; height: 20px;"
            >
              ‌
            </span>
          </span>
        </div>
        <span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton sc-jRQAMF eDrjwW css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
